cmd/kubelet/app/server_windows.go

```diff
@@ -20,66 +20,14 @@ limitations under the License.
 package app
 
 import (
-	"fmt"
-	"os/user"
+	"errors"
 
 	"golang.org/x/sys/windows"
 )
 
-func isAdmin() (bool, error) {
-	// Get current user
-	u, err := user.Current()
-	if err != nil {
-		return false, fmt.Errorf("error retrieving current user: %s", err)
-	}
-	// Get IDs of group user is a member of
-	ids, err := u.GroupIds()
-	if err != nil {
-		return false, fmt.Errorf("error retrieving group ids: %s", err)
-	}
-
-	// Check for existence of BUILTIN\ADMINISTRATORS group id
-	for i := range ids {
-		// BUILTIN\ADMINISTRATORS
-		if "S-1-5-32-544" == ids[i] {
-			return true, nil
-		}
-	}
-	return false, nil
-}
-
 func checkPermissions() error {
-	//https://github.com/golang/go/issues/28804#issuecomment-505326268
-	var sid *windows.SID
-	var userIsAdmin bool
-
-	// https://docs.microsoft.com/en-us/windows/desktop/api/securitybaseapi/nf-securitybaseapi-checktokenmembership
-	err := windows.AllocateAndInitializeSid(
-		&windows.SECURITY_NT_AUTHORITY,
-		2,
-		windows.SECURITY_BUILTIN_DOMAIN_RID,
-		windows.DOMAIN_ALIAS_RID_ADMINS,
-		0, 0, 0, 0, 0, 0,
-		&sid)
-	if err != nil {
-		return fmt.Errorf("error while checking for elevated permissions: %s", err)
-	}
-
-	//We must free the sid to prevent security token leaks
-	defer windows.FreeSid(sid)
-	token := windows.Token(0)
-
-	userIsAdmin, err = isAdmin()
-	if err != nil {
-		return fmt.Errorf("error while checking admin group membership: %s", err)
-	}
-
-	member, err := token.IsMember(sid)
-	if err != nil {
-		return fmt.Errorf("error while checking for elevated permissions: %s", err)
-	}
-	if !member {
-		return fmt.Errorf("kubelet needs to run with administrator permissions. Run as admin is: %t, User in admin group: %t", member, userIsAdmin)
+	if !windows.GetCurrentProcessToken().IsElevated() {
+		return errors.New("kubelet needs to run with elevated permissions!")
```

Comment:
Would it be useful to output who kubelet is running as here for logging?

Reply:
I thought about logging who the current user was (https://github.com/kubernetes/kubernetes/pull/108146#discussion_r807424845) but it is so simple enough to get with powershell I'm not sure it will add much value.

```
Get-Process -Name kubelet -IncludeUserName
```

Reply:
+1 on getting the user name in logs for remote troubleshooting. Getting on the node to run powershell may be too involved.

Reply:
@marosset Could you add the log? Otherwise this looks good. Thanks!

Reply:
Updated to log the current user.
(Sorry I forgot about this...)
PTAL